docs/addon-guide.md

```diff
@@ -0,0 +1,193 @@
+must report its status using the task (addon) report:
+1. Report addon started: `POST /tasks/:id/report` Task.State=Running.
+2. Report addon activity & progress: `PUT /tasks/:id/report` (repeat as needed).
+3. Report addon completed: `PUT /tasks/:id/report` Task.State=(Succeeded|Failed)
+
+The knowledge about an application consists of:
+- Application (model).
+- Source repository.
+- Tags.
+- Application Facts (key/value pair).
+- Files (stored in the application *bucket*).
+- Assessments.
+
+Modernization assets may be stored in:
+- Source repository.
+- Application Facts (key/value pair).
+- Files (stored in the application *bucket*).
+
+## Definition
+
+An Addon is defined and registered with the Hub using a Custom Resource (CR).
+
+* **image** - The addon image.
+* **resources** - An *(optional)* standard k8s pod container [resources](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers) specification.
+* **mounts** - An optional list of volumes to be mounted in the addon pod.
+    * **claim** - PVC name.
+    * **name** - The name of the directory in `/mnt` to mount the volume.
+
+Example:
+```
+apiVersion: tackle.konveyor.io/v1alpha1
+kind: Addon
+metadata:
+  name: admin
+  namespace: konveyor-tackle
+spec:
+  image: quay.io/konveyor/tackle2-addon:latest
+  mounts:
+  - claim: tackle-maven-volume-claim
+    name: m2
+  resources:
+    requests:
+      cpu: 50m
+      memory: 50Mi
+```
+
+## Runtime (pod)
+
+### Environment
+
+The pod is started with the following environment variables:
+  - **ADDON_WORKING_DIR** - Path to addon working directory.
+  - **HUB_BASE_URL** - Base URL for the hub API.
+  - **TOKEN** - An authentication token.
+  - **TASK** - An associated task ID.
+
+The bucket volume is mounted in `/buckets`.
+
+The addon container must have the default entry point set.
+
+### Go Binding
+
+The addon is expected to integrate with the Hub inventory using the
+Hub REST API. The secret includes an auth token. For convenience,
+the Hub (package)provides a `Go` binding simplifies integration.
+
+#### Run API.
+
+The Run() method simplifies/wraps addon execution.
+Provides addon (task) reporting of:
+ - Addon started.
+ - Addon succeeded when no error is returned.
+ - Addon failed when error is returned.
+ - Addon failed when panic detected.
+
+Recommended addon _main_.
+
+```
+package main
+
+import (
+    hub "github.com/konveyor/tackle2-hub/addon"
+    "github.com/konveyor/tackle2-hub/api"
+)
+
+var (
+    // hub integration. 
+    addon = hub.Addon
+    Log   = hub.Log
+)
+
+type SoftError = hub.SoftError
+
+//
+// main
+func main() {
+    addon.Run(func() (err error) {
+        // 
+        // Get the addon data associated with the task. 
+        d := &Data{}
+        _ = addon.DataWith(d)
+        if err != nil {
+            return
+        }
+        // 
+        // EXECUTE ADDON HERE.
+    }
+}
+```
+
+#### Tasks.
```

Comment:
I think that makes sense. It there's a stable link we can provide to the most up to date code rather than trying to constantly try to remember to update the code in the doc that would be ideal.

Reply:
Agreed.
This is in progress.